libsoftwarecontainer/unit-test/networkgatewayfunctions_unittest.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include "gateway/network/networkgatewayfunctions.h"
+#include "gateway/network/networkgateway.h"
+
+using namespace softwarecontainer;
+
+struct testParams
+{
+    uint32_t    netmask;
+    std::string gatewayIP;
+    int32_t     containerID;
+    bool        expectSuccess;
+};
+
+class NetworkGatewayFunctionsTests : public  ::testing::TestWithParam<testParams>
+{
+public:
+    NetworkGatewayFunctions ngf;
+    testParams testparams;
+
+    void SetUp() override
+    {
+        testparams = GetParam();
+    }
+};
+
+/*
+ * This data is fed to the NetworkGatewayFunctions tests
+ *
+ * to be able to understand the test it should be aware of following facts
```

Comment:
Revise wording here. Use punctuation and capital letters in beginning of sentences. For example, now this looks like a copy paste error kind of deal.